packages/jellyfish-api-core/__tests__/category/loan/getLoanInfo.test.ts

```diff
@@ -0,0 +1,564 @@
+
+    const tempAddress = await testing.generateAddress()
+    await testing.token.dfi({ amount: 10000, address: tempAddress })
+    await testing.container.generate(1)
+    await testing.rpc.loan.depositToVault({
+      vaultId: vault1Id,
+      from: tempAddress,
+      amount: '10000@DFI'
+    })
+    await testing.container.generate(1)
+
+    const extendedStartingData: GetLoanInfoResult = {
+      ...startingData,
+      currentPriceBlock: new BigNumber(108),
+      nextPriceBlock: new BigNumber(114),
+      defaults: {
+        ...startingData.defaults,
+        scheme: 'scheme1'
+      },
+      totals: {
+        ...startingData.totals,
+        schemes: new BigNumber(1),
+        collateralTokens: new BigNumber(1),
+        collateralValue: new BigNumber(10000),
+        openVaults: new BigNumber(1),
+        loanTokens: new BigNumber(1)
+      }
+    }
+
+    { // before
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual(extendedStartingData)
+    }
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vault1Id,
+      amounts: '2@TSLA'
+    })
+    await testing.container.generate(1)
+
+    { // After loan taken
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual({
+        ...extendedStartingData,
+        totals: {
+          ...extendedStartingData.totals,
+          loanValue: expect.any(BigNumber)
+        }
+      })
+      // loan value will include some interest
+      expect(data.totals.loanValue.gt(10)).toBeTruthy() // 2 * 5
+      expect(data.totals.loanValue.lt(10.0001)).toBeTruthy()
+    }
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vault1Id,
+      amounts: '1@TSLA'
+    })
+    await testing.container.generate(1)
+
+    { // After loan taken
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual({
+        ...extendedStartingData,
+        totals: {
+          ...extendedStartingData.totals,
+          loanValue: expect.any(BigNumber)
+        }
+      })
+      // loan value will include some interest
+      expect(data.totals.loanValue.gt(15)).toBeTruthy() // 3 * 5
+      expect(data.totals.loanValue.lt(15.0001)).toBeTruthy()
+    }
+  })
+
+  it('should return with updated loan related governance variables (delayed effect, currentpriceblock must not be affected)', async () => {
+    { // before
+      const data = await testing.rpc.loan.getLoanInfo()
+      expect(data).toStrictEqual(startingData)
+    }
+
+    await testing.rpc.masternode.setGov({
+      ORACLE_BLOCK_INTERVAL: 7
+    })
+    await testing.generate(1)
+
+    { // after govvar set
+      const data = await testing.rpc.loan.getLoanInfo()
+      console.log('after set:', data)
+      expect(data).toStrictEqual({
+        ...startingData,
+        defaults: {
+          ...startingData.defaults,
+          fixedIntervalBlocks: new BigNumber(6)
+        }
+      })
+    }
+
+    await testing.container.waitForBlockHeight(111)
```

Comment:
interesting behavior. new `ORACLE_BLOCK_INTERVAL` kicks in at 108. it was set at 105.
after kick in it modifies the `nextPriceBlock` to `112` and `currentPriceBlock` to 105. not sure if that's the expected behavior. few questions

1. should `setGov` activate the variable at next block ?
2. after activation of `ORACLE_BLOCK_INTERVAL`, should it be `105 + 7 = 112 `or `108+7 = 115` ?

I know that this is exactly not related to `getLoanInfo`. But the behavior seems a bit odd.

Reply:
Yes, that's expected behavior for now.

1. After few consideration and iteration, cpp team want the `blockheight % ORACLE_BLOCK_INTERVAL` must always zero, to best accommodate other stuff.
2. At the same time, to minimize the impact on `nextPriceBlock` that is already disclosed to user. It will always kicks in after whatever `nextPriceBlock` is during the block of setGov called.

Combining these points, resulted some weird price block changes but seems to be the best we had so far